src/jon_mikkel_ex/ex04/walker.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+__author__ = "Jon-Mikkel Korsvik"
+__email__ = "jonkors@nmbu.no"
+
+from random import getrandbits
+from pprint import pprint
+
+
+class Walker:
+    """
+    Walker class, with methods for checking position and moving.
+    Has its own method for finishing a trip home.
+
+    :param the_club: initial position of the walker
+    :param home: position of the walker's home
+    """
+    def __init__(self, the_club, home):
+        self.position = the_club
+        self.home = home
+        self.steps = 0
+
+    def move(self):
+        """
+        Uses a really efficient cointoss-  bool(getrandbits(1)) gives true or
+        false. Registers each move as a step.
+        :return: New position after move
+        """
+        if bool(getrandbits(1)):
+            self.position += 1
+        else:
+            self.position -= 1
+        self.steps += 1
+        return self.position
+
+    def is_at_home(self):
+        return self.position == self.home
+
+    def get_position(self):
+        return self.position
+
+    def get_steps(self):
+        return self.steps
+
+    def make_the_trip(self):
+        while True:
+            self.move()
+            if self.is_at_home():
+                return self.get_steps()
```

Comment:
In this loop, remove the `if` statement, and put the exit condition directly after `while`: `while not self.is_at_home(): ....`

Reply:
Then put the return outside the loop.